test/argmaxpool-microkernel-tester.h

```diff
@@ -290,3 +292,890 @@ class ArgMaxPoolMicrokernelTester {
   size_t output_stride_{0};
   size_t iterations_{3};
 };
+
+void ArgmaxPoolUnipassTest(
+  xnn_f32_argmaxpool_unipass_ukernel_fn ukernel,
+  const std::string kernel_name,
+  size_t _pooling_elements,
+  size_t _primary_tile,
+  size_t _incremental_tile,
+  size_t _channels,
+  size_t _input_offset)
+{
+  ArgMaxPoolMicrokernelTester tester;
+  tester.pooling_elements(_pooling_elements);
+  tester.pooling_tile(_primary_tile, (_incremental_tile != 0) ? _incremental_tile : 0);
+  tester.channels(_channels);
+  if (_input_offset != 0) {
+    tester.input_offset(_input_offset);
+  }
+  if (kernel_name.find("scalar") != std::string::npos) {
+    tester.Test(ukernel, ArgMaxPoolMicrokernelTester::Variant::Native);
+  }
+  else {
+    tester.Test(ukernel);
+  }
+}
+
+void ArgmaxPoolMultipassTest(
+  xnn_f32_argmaxpool_multipass_ukernel_fn ukernel,
+  const std::string kernel_name,
+  size_t _pooling_elements,
+  size_t _primary_tile,
+  size_t _incremental_tile,
+  size_t _channels,
+  size_t _input_offset)
+{
+  ArgMaxPoolMicrokernelTester tester;
+  tester.pooling_elements(_pooling_elements);
+  tester.pooling_tile(_primary_tile, (_incremental_tile != 0) ? _incremental_tile : 0);
+  tester.channels(_channels);
+  if (_input_offset != 0) {
+    tester.input_offset(_input_offset);
+  }
+  if (kernel_name.find("scalar") != std::string::npos) {
+    tester.Test(ukernel, ArgMaxPoolMicrokernelTester::Variant::Native);
```

Comment:
Sorry for the churn, this is unused, we should remove it, I've sent a PR to do that.

If you see something like this that doesn't make sense and is unused, please clean it up. We shouldn't need to use the stringified name of kernels for anything.